From 71d8852bb5bbf0b9325387f263883df5d4e289b4 Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Mon, 19 Nov 2012 16:54:42 -0800
Subject: [PATCH 1214/1357] MR1: Backport MAPREDUCE-4800. Remove Unnecessary code from MapTaskStatus.

Reason: Part of the fix for CDH-8955 (Mapper progress is not reported)
Ref: CDH-8955
Author: Karthik Kambatla
---
 .../org/apache/hadoop/mapred/MapTaskStatus.java    |   17 +++--------------
 src/mapred/org/apache/hadoop/mapred/Task.java      |    2 +-
 2 files changed, 4 insertions(+), 15 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapred/MapTaskStatus.java b/src/mapred/org/apache/hadoop/mapred/MapTaskStatus.java
index fc78da2..2607771 100644
--- a/src/mapred/org/apache/hadoop/mapred/MapTaskStatus.java
+++ b/src/mapred/org/apache/hadoop/mapred/MapTaskStatus.java
@@ -26,7 +26,6 @@ import java.io.IOException;
 class MapTaskStatus extends TaskStatus {
 
   private long mapFinishTime;
-  private long sortFinishTime;
   
   public MapTaskStatus() {}
 
@@ -49,10 +48,10 @@ class MapTaskStatus extends TaskStatus {
   @Override
   void setFinishTime(long finishTime) {
     super.setFinishTime(finishTime);
-    if (mapFinishTime == 0) {
-      mapFinishTime = finishTime;
+    // set mapFinishTime if it hasn't been set before
+    if (getMapFinishTime() == 0) {
+      setMapFinishTime(finishTime);
     }
-    setSortFinishTime(finishTime);
   }
   
   @Override
@@ -74,16 +73,6 @@ class MapTaskStatus extends TaskStatus {
   void setMapFinishTime(long mapFinishTime) {
     this.mapFinishTime = mapFinishTime;
   }
-
-  @Override
-  public long getSortFinishTime() {
-    return sortFinishTime;
-  }
-
-  @Override
-  void setSortFinishTime(long sortFinishTime) {
-    this.sortFinishTime = sortFinishTime;
-  }
   
   @Override
   synchronized void statusUpdate(TaskStatus status) {
diff --git a/src/mapred/org/apache/hadoop/mapred/Task.java b/src/mapred/org/apache/hadoop/mapred/Task.java
index 05255ec..cc65143 100644
--- a/src/mapred/org/apache/hadoop/mapred/Task.java
+++ b/src/mapred/org/apache/hadoop/mapred/Task.java
@@ -580,7 +580,7 @@ abstract public class Task implements Writable, Configurable {
     
     public float getProgress() {
       return taskProgress.getProgress();
-    };
+    }
     
     public void progress() {
       // indicate that progress update needs to be sent
-- 
1.7.0.4

